pina/model/avno.py

```diff
@@ -0,0 +1,164 @@
+                                    inner_size, n_layers, func)
+        self.points_size = points_size
+        self.inner_size = inner_size
+        self.input_features = input_features
+        self.points_label = points_label
+        self.features_label = features_label
+
+    def forward(self, batch):
+        """Forward pass of the layer."""
+        points_tmp = concatenate([
+            batch.extract(f"{self.points_label}_{i}")
+            for i in range(self.points_size)
+        ],
+                                 axis=2)
+        features_tmp = concatenate([
+            batch.extract(f"{self.features_label}_{i}")
+            for i in range(self.input_features)
+        ],
+                                   axis=2)
+        new_batch = concatenate((features_tmp, points_tmp), dim=2)
+        new_batch = self._lifting(new_batch)
+        return [new_batch, points_tmp]
+
+    def net(self):
+        """Returns the net"""
+        return self._lifting
+
+
+class Integral_Net(nn.Module):
+    """
+    The PINA implementation of the integral layer of the AVNO.
+
+    :param int inner_size: number of neurons in the hidden layer(s).
+    :param int n_layers: number of hidden layers. Default is 4.
+    :param func: the activation function to use. Default to nn.GELU.
+    
+    """
+
+    def __init__(self, inner_size, n_layers, func):
+        super().__init__()
+        self._nn = nn.Sequential(
+            *[AVNOBlock(inner_size, func) for _ in range(n_layers)])
+
+    def forward(self, batch):
+        """Forward pass of the layer."""
+
+        new_batch, points_tmp = batch
+        new_batch = self._nn(new_batch)
+        return [new_batch, points_tmp]
+
+    def net(self):
+        """Returns the net"""
+        return self._nn
+
+
+class Projection_Net(nn.Module):
+    """
+    The PINA implementation of the projection 
+        layer of the AVNO.
+
+    :param int inner_size: number of neurons in the hidden layer(s).
+    :param int points_size: the dimension of the domain of the functions.
+    :param int output_features: The number of output components of the model.
+    :param int n_layers: number of hidden layers. Default is 4.
+    :param func: the activation function to use. Default to nn.GELU.
+
+    """
+
+    def __init__(self, inner_size, points_size, output_features, n_layers,
+                 func):
+        super().__init__()
+        self._projection = FeedForward(inner_size + points_size,
+                                       output_features, inner_size, n_layers,
+                                       func)
+
+    def forward(self, batch):
+        """Forward pass of the layer."""
+        new_batch, points_tmp = batch
+        new_batch = concatenate((new_batch, points_tmp), dim=2)
+        new_batch = self._projection(new_batch)
+        return new_batch
+
+    def net(self):
+        """Returns the net"""
+        return self._projection
+
+
+class AVNO(KernelNeuralOperator):
```

Comment:
Extended name

Reply:
Agree

Reply:
You mean that it should be AveragingNeuralOperator?